presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowPageUtils.java

```diff
@@ -0,0 +1,933 @@
+        }
+        else if (vector instanceof DecimalVector) {
+            return buildBlockFromDecimalVector((DecimalVector) vector, type);
+        }
+        else if (vector instanceof NullVector) {
+            return buildBlockFromNullVector((NullVector) vector, type);
+        }
+        else if (vector instanceof TimeStampMicroVector) {
+            return buildBlockFromTimeStampMicroVector((TimeStampMicroVector) vector, type);
+        }
+        else if (vector instanceof TimeStampMilliVector) {
+            return buildBlockFromTimeStampMilliVector((TimeStampMilliVector) vector, type);
+        }
+        else if (vector instanceof Float4Vector) {
+            return buildBlockFromFloat4Vector((Float4Vector) vector, type);
+        }
+        else if (vector instanceof Float8Vector) {
+            return buildBlockFromFloat8Vector((Float8Vector) vector, type);
+        }
+        else if (vector instanceof VarCharVector) {
+            if (type instanceof CharType) {
+                return buildCharTypeBlockFromVarcharVector((VarCharVector) vector, type);
+            }
+            else if (type instanceof TimeType) {
+                return buildTimeTypeBlockFromVarcharVector((VarCharVector) vector, type);
+            }
+            else {
+                return buildBlockFromVarCharVector((VarCharVector) vector, type);
+            }
+        }
+        else if (vector instanceof VarBinaryVector) {
+            return buildBlockFromVarBinaryVector((VarBinaryVector) vector, type);
+        }
+        else if (vector instanceof DateDayVector) {
+            return buildBlockFromDateDayVector((DateDayVector) vector, type);
+        }
+        else if (vector instanceof DateMilliVector) {
+            return buildBlockFromDateMilliVector((DateMilliVector) vector, type);
+        }
+        else if (vector instanceof TimeMilliVector) {
+            return buildBlockFromTimeMilliVector((TimeMilliVector) vector, type);
+        }
+        else if (vector instanceof TimeSecVector) {
+            return buildBlockFromTimeSecVector((TimeSecVector) vector, type);
+        }
+        else if (vector instanceof TimeStampSecVector) {
+            return buildBlockFromTimeStampSecVector((TimeStampSecVector) vector, type);
+        }
+        else if (vector instanceof TimeMicroVector) {
+            return buildBlockFromTimeMicroVector((TimeMicroVector) vector, type);
+        }
+        else if (vector instanceof TimeStampMilliTZVector) {
+            return buildBlockFromTimeMilliTZVector((TimeStampMilliTZVector) vector, type);
+        }
+        else if (vector instanceof ListVector) {
+            return buildBlockFromListVector((ListVector) vector, type);
+        }
+
+        throw new UnsupportedOperationException("Unsupported vector type: " + vector.getClass().getSimpleName());
+    }
+
+    public static Block buildBlockFromEncodedVector(FieldVector encodedVector, FieldVector dictionary)
```

Comment:
can't we accept `Dictionary` type itself here.

Reply:
done